src/i_winmusic.c

```diff
@@ -1488,34 +1502,35 @@ static void I_WIN_SetMusicVolume(int volume)
 
     volume_factor = sqrtf((float)volume / 15);
 
-    update_volume = (song.file != NULL);
+    EnterCriticalSection(&CriticalSection);
+
+    win_midi_state = STATE_VOLUME;
```

Comment:
If the volume is adjusted while the song is paused, it will begin playing. So maybe add `if (win_midi_state == STATE_PLAYING)` here.

Reply:
I don't think we need STATE_VOLUME. Even if the song is paused, we still want to update the volume when playback starts.